app/models/course/experience_points_record.rb

```diff
@@ -74,4 +75,45 @@ def set_awarded_attributes
     self.awarded_at ||= Time.zone.now
     self.awarder ||= User.stamper
   end
+
+  def validate_limit_exp_points_on_association
+    return if manually_awarded?
+
+    case specific.actable
+    when Course::Assessment::Submission
+      return unless specific.assessment
```

Comment:
previously it has error, I thought it's because assessment's not tied to submission but apparently it's because the points awarded is nil. After doing the fix, I forgot to delete this. Really sorry, now it's deleted already